### PR TITLE
Graphql Types

### PR DESCRIPTION
**Pull request by DerekStride**
This PR builds on #9 to add the GraphQL Types for our models.

We need to add more seeding tasks so we can set up and test the relationships between the data models.

### CODE REVIEW

File: app/graph/types/comment_type.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CommentType = GraphQL::ObjectType.define do
+  name 'Comment'
+  description 'A comment that a user left on a post.'
```
Comment (justinkrol) [inline app/graph/types/comment_type.rb]: 'A comment a user left on some _content_'

**Review by justinkrol — APPROVED** [review]
Just the one change on the description for `CommentType`

**Comment by ismail-syed** [comment]
Having issues making queries using GraphiQL with this changes. Do I need to seed my local db for this to work?

```
╰─$ rails s
=> Booting Puma
=> Rails 5.1.4 application starting in development
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.10.0 (ruby 2.4.1-p111), codename: Russell's Teapot
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
Use Ctrl-C to stop
Started POST "/graphql" for 127.0.0.1 at 2017-10-02 22:19:08 -0400
   (0.4ms)  SET  @@SESSION.sql_mode = CONCAT(CONCAT(@@sql_mode, ',STRICT_ALL_TABLES'), ',NO_AUTO_VALUE_ON_ZERO'),  @@SESSION.sql_auto_is_null = 0, @@SESSION.wait_timeout = 2147483
   (2.2ms)  SELECT `schema_migrations`.`version` FROM `schema_migrations` ORDER BY `schema_migrations`.`version` ASC
Processing by GraphqlController#query as */*
  Parameters: {"query"=>"# Welcome to GraphiQL\n#\n# GraphiQL is an in-browser tool for writing, validating, and\n# testing GraphQL queries.\n#\n# Type queries into this side of the screen, and you will see intelligent\n# typeaheads aware of the current GraphQL type schema and live syntax and\n# validation errors highlighted within the text.\n#\n# GraphQL queries typically start with a \"{\" character. Lines that starts\n# with a # are ignored.\n#\n# An example GraphQL query might look like:\n#\n#     {\n#       field(arg: \"value\") {\n#         subField\n#       }\n#     }\n#\n# Keyboard shortcuts:\n#\n#       Run Query:  Ctrl-Enter (or press the play button above)\n#\n#   Auto Complete:  Ctrl-Space (or just start typing)\n#\n\nquery {\n  simple\n \n}", "variables"=>nil, "graphql"=>{"query"=>"# Welcome to GraphiQL\n#\n# GraphiQL is an in-browser tool for writing, validating, and\n# testing GraphQL queries.\n#\n# Type queries into this side of the screen, and you will see intelligent\n# typeaheads aware of the current GraphQL type schema and live syntax and\n# validation errors highlighted within the text.\n#\n# GraphQL queries typically start with a \"{\" character. Lines that starts\n# with a # are ignored.\n#\n# An example GraphQL query might look like:\n#\n#     {\n#       field(arg: \"value\") {\n#         subField\n#       }\n#     }\n#\n# Keyboard shortcuts:\n#\n#       Run Query:  Ctrl-Enter (or press the play button above)\n#\n#   Auto Complete:  Ctrl-Space (or just start typing)\n#\n\nquery {\n  simple\n \n}", "variables"=>nil}}
Completed 500 Internal Server Error in 32ms (ActiveRecord: 0.0ms)



NotImplementedError (schema contains Interfaces or Unions, so you must define a `resolve_type -> (obj, ctx) { ... }` function):

app/graph/app_schema.rb:1:in `<top (required)>'
app/controllers/graphql_controller.rb:4:in `query'
```

screen shot 2017-10-02 at 10 22 08 pm

`NotImplementedError`; presuming everything isn't fully implemented?

**Comment by DerekStride** [comment]
I forced pushed a fix, let me know if it worked

**Comment by ismail-syed** [comment]
@DerekStride It's a bit better now. I'm getting null responses, is that expected for now? 

screen shot 2017-10-04 at 9 38 11 pm

**Comment by DerekStride** [comment]
@ismail-syed In graphql you typically is a global id as the identifier (e.g. https://github.com/rails/globalid). Try this

```graphql
query {
  user(id: "gid://webscale/User/2") {
    email
  }
}
```

**Review by ismail-syed — APPROVED** [review]

**Pull request merged**